### ISSUE TITLE
[Hunter][Spell] Kill Command

### ISSUE THREAD

**Issue by Zabbil** [dalaranwow/dalaran-wow#4676]
[//]: # (INFORMATION ABOUT TAGS)
[//]: # (The black and yellow tag is unique. There can only be one label assigned to them - one black and one yellow)
[//]: # (Priority - ... tags are assigned by team members)
[//]: # (Your issue will be reviewed by testers and the developers will fix it in the future.)
[//]: # (Only once the issue got the label [Fixed - On Live Server] you can find the fix on your Gameserver.)
[//]: # (You can increase the relevance of your issue by adding a thumb-up emote http://i.imgur.com/RFouVVi.png)

[//]: # (***************************************************************)
[//]: # (** !!!!!!!!!!!!!DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED!!!!!!!!!!!! **)
[//]: # (***************************************************************)

**Current behaviour**: Tell us what happens in detail (Where, When, What, Why, Who)

Casting Kill Command can proc melee weapon skill up. 

**Expected behaviour**: Tell us what should happen instead

Casting Kill Command should not result in weapon skill up.

**Steps to reproduce the problem**:

1. Cast Kill Command, observe that you get skill up on your currently equipped melee weapon. 
(Seems to have higher chance to proc when you have low weapon skill. Same mechanic as when a high level character with low weapon skill has a greater chance to proc skill up when using weapon)
2. 
3. 

**Include proofs for this behaviour**
![cast kill command](https://cloud.githubusercontent.com/assets/27431381/25037178/d95d0fc2-20f7-11e7-91e3-f4983c4f77eb.JPG)
![melee weapon skill up](https://cloud.githubusercontent.com/assets/27431381/25037187/e43714ce-20f7-11e7-8a0a-d7ed5f828851.JPG)

You can research this on several pages like http://wowhead.com or http://wowwiki.wikia.com or http://www.youtube.com/
Issues without proofs will be closed and tag On Hold will be added until the proof is provided.

**Include the ID for the game objects, npcs (creatures,pets,minions), spells, items, quests, instances, zones, achievements, skills**
http://wotlk.openwow.com/spell=34026
You can research this on several pages like http://wotlk.openwow.com or http://wowhead.com/

**Include Screenshots from the issue if necessary**

Can this possible have any interaction with the Auto Shot bug? What I mean is; can the “melee” ability of Kill Command in any way interact and disturb Auto Shot sequence?

I have no proof, nor was I able to reproduce this. But I am wondering if this might be something that’s worth looking in to?

Auto Shot issue #834 and #4653

**Comment by TeridanDW** [comment]
Confirmed. Thank you for the report @Zabbil .
I would like to add 1 more issue here:
 - Kill command should NOT generate any threat for the hunter, however currently it does.

Proof:
http://www.wowhead.com/spell=34026/kill-command

> Flags	
> Cannot be used while shapeshifted
> Cannot be dodged, parried or blocked
> **_Generates no threat_**

Example from my TC:
![image](https://cloud.githubusercontent.com/assets/6797611/25038065/561f31b8-2105-11e7-961d-5227aec80e20.png)
![image](https://cloud.githubusercontent.com/assets/6797611/25038068/5c8280be-2105-11e7-9744-aa9253a0de06.png)

**Comment by Rushor** [comment]
Game/Unit: Player - Do not UpdateCombatSkills if target is a pet 

* closes: https://github.com/dalaranwow/dalaran-wow/issues/4676
* commit: 33cc40b4eff8be6e3eff8e4f4a062a01d1041720

The issue was fixed by a developer and will now be tested on the Public Test Realm (PTR). The changes/fixes will be applied with the NEXT changelog on the liverealm.
This means that the issue exists in game at the moment i wrote this message.
Only once the issue was marked with "Fixed - On Live Server", you can find the fix on you server.
Please watch http://www.dalaran-wow.com/news/ to see when the changelog and the fix was applied.

**Comment by TeridanDW** [comment]
After we tests on PTR:
Issue:

> Casting Kill Command can proc melee weapon skill up.

is fixed.

> Kill command should NOT generate any threat for the hunter, however currently it does.

is still a problem.

On the screenshot you can pet attacking the target and me casting few Kill Commands only (no other interactions):
![image](https://user-images.githubusercontent.com/6797611/31833693-155732d4-b5d4-11e7-9ce5-b0f0dee82342.png)

**Comment by TeridanDW** [comment]
2-nd issue is still happening.
